Implemented CheckBanned and Unban (Attempt two)

jonfriesen opened the pull request:
Built two additional commands (unban, checkbanned) and made some changes to
the existing ban command.

**Changes**

ban now shows notification to admin user in bad
custom message if admin tries to ban themselves

**Additions**

unban command: _/unban username_
checkbanned command: _/checkbanned_ or _/checkbanned user_ (shows all banned users or
if a specific users banned status)
I built this PR based on #1075 and it should be noted that the existing
ban command already works on offline users.

I've also added unit tests for both unban and checkbanned commands.

This is my second attempt at posting this. The first time went horrible because I can't GIT. This time I've used an up-to-date branch so my PR shouldn't be stale. Also all commits are squashed instead of a ton pushing.

Hope this is up to your standards, if I can do anything better regarding code style, git pushing, etc I would greatly appreciate the advice!

jonfriesen committed:
Implemented IsBanned and Unban

Built two additional commands (unban, isbanned) and made some changes to
the existing ban command.

Changes

ban now shows notification to admin user in bad
custom message if admin tries to ban themselves
Additions

unban command: /unban username
isbanned command: /isbanned or /isbanned user (shows all banned users or
if a specific user is banned)
I built this PR based on #1075 and it should be noted that the existing
ban command already works on offline users.

hnrkndrssn commented:
Looks good to me

samandmoore commented on JabbR/Hubs/Chat.cs:
```diff
@@ -1199,6 +1199,38 @@ void INotificationService.BanUser(ChatUser targetUser)
+
+        void INotificationService.IsBanned()
+        {
+            // Get all users that are banned
```
is this comment really useful?

samandmoore commented:
this looks great. it would be swell to also have some tests added for it though: like this. https://github.com/JabbR/JabbR/blob/dev/JabbR.Tests/CommandManagerFacts.cs#L2365

jonfriesen committed:
Added unit tests for unban and isbanned

Simple unit tests for Unban to handle
- Missing username
- Non-Admin use of command
- Unban nonexistent user
- Cannot Unban another Admin
- Cannot Unban yourself
- Unbanning a banned user

for IsBanned to handle
- Get banned user list
- Non-admin user
- Non-existent user

Updated test when trying to ban yourself to appropriate exception text

Updated LanguageResources message for Unban_UserRequired

jonfriesen commented:
@samandmoore, thanks for the advice! Added unit tests for both commands. I'm still grasping how mocking works, so (I think) I'm a little light in that area still.

samandmoore commented:
lgtm with the tests.

davidfowl commented on JabbR/Services/INotificationService.cs:
```diff
@@ -39,6 +39,9 @@ public interface INotificationService
         void BanUser(ChatUser targetUser);
+        void UnbanUser(ChatUser targetUser);
+        void IsBanned(ChatUser targetUser);
+        void IsBanned();
```
This name needs to change across the board. This should be CheckBanned

jonfriesen committed:
Name change: isbanned > checkbanned

All instances of isbanned are now checkbanned for commands, with
appropriate casing for classes.
